src/brs/http/GetAccountAssets.java

```diff
@@ -0,0 +1,54 @@
+package brs.http;
+
+import brs.Account;
+import brs.BurstException;
+import brs.services.AccountService;
+import brs.services.ParameterService;
+import brs.util.Convert;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+
+import javax.servlet.http.HttpServletRequest;
+
+import static brs.http.common.Parameters.ACCOUNT_PARAMETER;
+import static brs.http.common.ResultFields.*;
+
+public final class GetAccountAssets extends APIServlet.JsonRequestHandler {
+
+    private final ParameterService parameterService;
+    private final AccountService accountService;
+
+    GetAccountAssets(ParameterService parameterService, AccountService accountService) {
+        super(new APITag[] {APITag.ACCOUNTS}, ACCOUNT_PARAMETER);
+        this.parameterService = parameterService;
+        this.accountService = accountService;
+    }
+
+    @Override
+    JsonElement processRequest(HttpServletRequest req) throws BurstException {
+
+        Account account = parameterService.getAccount(req);
+
+        JsonObject response = new JsonObject();
+
+        JsonArray assetBalances = new JsonArray();
+        JsonArray unconfirmedAssetBalances = new JsonArray();
+
+        for (Account.AccountAsset accountAsset : accountService.getAssets(account.getId(), 0, -1)) {
```

Comment:
This needs to have a limit on it. Can't remember whether it's in V2 or V3, but the API responses are by default limited to 100 returned items. What if an account has 1000+ assets? This call will take a non-trivial amount of time. The call needs to have `firstIndex` and `lastIndex` parameters added.

Reply:
good points. I not thought more deeply about this API call, I just copy logic from original code